### PR TITLE
refactor: move macOS default app logic to dedicated class 

### PR DESCRIPTION
**Pull request by nevack**
~~Supersedes #6099 as GitHub doesn't support parent/dependent PR as a concept.
Creating this before #6099 is merged to gather feedback and to run CI checks.~~
Parent PR is merged.

### CODE REVIEW

**Comment by nevack** [comment]
@Coeur JFYI, only the 2nd commit is worth checking here.

**Review by cleberpereiradasilva — APPROVED** [review]
lgtm

**Comment by ckerr** [comment]
needs another review from @transmission/contributors

File: macosx/DefaultAppHelper.mm
Comment (DevilDimon) [inline macosx/DefaultAppHelper.mm:46]: Are we using snake case now?

File: macosx/DefaultAppHelper.mm
Comment (DevilDimon) [inline macosx/DefaultAppHelper.mm:13]: Any reason why it's a freestading func and not a class method?

File: macosx/DefaultAppHelper.mm
Comment (DevilDimon) [inline macosx/DefaultAppHelper.mm:89]: completionHandler is nullable – this will crash if it's nil

File: macosx/DefaultAppHelper.mm
Comment (DevilDimon) [inline macosx/DefaultAppHelper.mm:103]: Ditto about the completionHandler being nil – also applies to the rest of the file

File: macosx/PrefsController.mm
Comment (DevilDimon) [inline macosx/PrefsController.mm:107]: It does not change so I suppose we can make it `readonly`

Comment (nevack) [reply macosx/PrefsController.mm]: Fixed

File: macosx/DefaultAppHelper.mm
Comment (nevack) [reply macosx/DefaultAppHelper.mm]: Fixed

File: macosx/DefaultAppHelper.mm
Comment (nevack) [reply macosx/DefaultAppHelper.mm]: Fixed by checking for `nil` every time before invoking.

File: macosx/DefaultAppHelper.mm
Comment (nevack) [reply macosx/DefaultAppHelper.mm:13]: Just used example from the docs of [UTType.h](https://github.com/alexey-lysiuk/macos-sdk/blob/main/MacOSX14.0.sdk/System/Library/Frameworks/UniformTypeIdentifiers.framework/Versions/A/Headers/UTType.h#L373-L402)

Any reason to prefer class method?

File: macosx/DefaultAppHelper.mm
Comment (nevack) [reply macosx/DefaultAppHelper.mm]: Fixed, I will look into actually checking naming conventions in ObjC++ with clang-format config.

**Review by Coeur — APPROVED** [review]

**Comment by Coeur** [comment]
Looks good, but please try to verify that if Transmission was already default app for torrents before this code change, then it's still the default app after this change, thank you. (asking, because we're changing the api, and I don't know if the new one is compatible with the old one)

**Comment by ckerr** [comment]
Writing down my thought process so the PR doesn't get stuck in limbo with everyone thinking it's someone else's turn to act :smile_cat: 

I see this is approved now, but the review starts with "Looks good, but please try to..." so it looks like there is still at least some manual testing to be done. I'll keep watching this PR for followup comments from either @nevack or @Coeur but will hold off on merging until someone gives the go-ahead

**Comment by Coeur** [comment]
@ckerr I just tested it successfully: upgrading from a previous version is working fine for default app detection. Good to merge

**Pull request merged**
